docker-app/qfieldcloud/core/admin.py

```diff
@@ -1103,6 +1121,14 @@ def has_module_permission(self, request: HttpRequest) -> bool:
         return False
 
 
+class LogEntryAdmin(ModelAdminEstimateCountMixin, BaseLogEntryAdmin):
+    list_filter = ["action"]
+    search_fields = [
+        *BaseLogEntryAdmin.search_fields,
+        "content_type__model",
+    ]
+
```

Comment:
Maybe should add some checks to avoid problems if upstream changes?

Performance gain:

The most expensive query comes from their filtering:

`    list_filter = ["action", ResourceTypeFilter, CIDFilter]` 
Where ResourceTypeFIlter ( https://github.com/jazzband/django-auditlog/blob/0cb3eb2b996cc14498dbdd94429169c64e23cfb9/auditlog/filters.py#LL5 ) gets slow on large dataset with many types.
![image](https://user-images.githubusercontent.com/21113500/225905871-8a283569-538b-4f54-868b-04f7700d5103.png)
(the red bar)

Instead could enable filtering the model type as search which then only takes time loading the specific filtered list. Or alternatively (TODO) provide a custom filter with options derived from available models list directly instead.

result:
![image](https://user-images.githubusercontent.com/21113500/225906418-70138efe-3538-4849-b57b-0b9c880d7b03.png)

Now next expensive queries are the two counts. With the `ModelAdminEstimateCountMixin` can complete avoid this:

Leading to only the actual list query taking up some time:
![image](https://user-images.githubusercontent.com/21113500/225906784-2f4eae2c-5502-4c6e-8496-9db2890ddc27.png)


Note this is done on a large  number of entries (+ 5M) but not complete data. Thus on production the queries still take way longer.

Reply:
Thanks for the input. What is not clear is how `    list_filter = ["action", ResourceTypeFilter, CIDFilter]` is expensive when no filter is applied.

Reply:
For the `ResourceTypeFilter` they need to query and join `django_content_type` over the whole dataset to get the filter options

![image](https://user-images.githubusercontent.com/21113500/226419966-c5da3fd2-a572-4aa4-aac2-a5cb5d51e055.png)

Because they do this expensive query in `def lookups(....)` method (s. here https://github.com/jazzband/django-auditlog/blob/0cb3eb2b996cc14498dbdd94429169c64e23cfb9/auditlog/filters.py#LL5 )

The best would be to just create an alternative filter by either hardcoding the lookup values or just all the model names from pg meta data (a bit like the estimate count) or retrieving this information somehow dynamically from the running python/django by getting all the admin registered model names?

Reply:
I see.

I did not follow the metadata part of your proposal, but I hope my suggestion below fits in the `somehow dynamically`.

Pseudo code suggestion:

```
from django_auditlog import ResourceTypeFilter

class QFieldCloudResourceTypeFilter(ResourceTypeFilter):
    def lookups(self, request, model_admin):
        qs = ContentType.objects.all().order_by("model")
        types = qs.values_list("id", "model")
        return types
```

Reply:
we can further make the function cacheable, as it is something that changes only if the app restarts. Actually there might be a django caching mechhanism for ContentTypes already, worth checking.

Reply:
> 

That looks very promising. Makes so much more sense than a distinct query over the whole dataset of logentries ... . We might also suggest this on their repo (upstream)?

Reply:
Yes makes sense

Reply:
Your pseudo code works exactly as is :D. Now the query for lookups only depends on model count not and quering the list entries (limited to page) is the main query

![image](https://user-images.githubusercontent.com/21113500/226430246-c5c0a236-9ef0-47a1-a2c7-769190b5c404.png)

only the filter options are too many. IMO that is even acceptable for internal use?